definitions/checks/check_hotfix_installed.rb

```diff
@@ -0,0 +1,92 @@
+class Checks::CheckHotfixInstalled < ForemanMaintain::Check
+  metadata do
+    label :check_hotfix_installed
+    description 'Check to verify if any hotfix installed on system'
+    preparation_steps do
+      Procedures::Packages::Install.new(:packages => %w[yum-utils])
+    end
+
+    confine do
+      feature(:downstream)
+    end
+
+    # param :version,
+    #      'Version for which repositories needs to be consider',
+    #      :required => true
+
+    # manual_detection
+  end
+
+  def run
+    # TODO: Remove this part as it is only for testing changes locally
+    @version = '6.2'
+    if feature(:downstream) && feature(:downstream).subscribed_using_activationkey?
+      skip 'Your system is subscribed using custom activationkey'
+    else
+      with_spinner('Checking for presence of hotfix(es). It may takes some time to verify.') do
+        hotfix_rpmlist = []
+
+        hotfix_rpmlist = find_hotfix_rpms_installed if feature(:downstream)
+        files_modifications = rpm_verify_command(@version)
+        assert(hotfix_rpmlist.empty? && files_modifications.empty?,
+               warning_message(hotfix_rpmlist, files_modifications))
+      end
+    end
+  end
+
+  private
+
+  def warning_message(hotfix_rpmlist, files_modifications)
+    message = "\n"
+    unless hotfix_rpmlist.empty?
+      message += "Found below HOTFIX rpm(s) applied on this system.\n"
+      message += hotfix_rpmlist
+    end
+    unless files_modifications.empty?
+      message += "Found #{files_modifications.length} file(s) modified on this system.\n"
+      if files_modifications.length > 10
+        message += 'Here, it shows only 10 records. For complete result, please check a log file.'
+      end
+      message += files_modifications[0..9].join("\n")
+    end
+    message += "\nBefore continuing upgrade, please verify above hotfix(es) details\n"
+    message
+  end
+
+  # only for downstream
+  def find_hotfix_rpms_installed
+    cmd = "rpm -qa 'HOTFIXRHBZ*'"
```

Comment:
this doesn't seem to match packages, that have `HOTFIX` state in release version (see https://bugzilla.redhat.com/show_bug.cgi?id=1642369 as an example for testing with some hotfix installed)

Reply:
It doesn't seem lik `rpm -qa` provides the capability. We could perhaps use the `find_installed_packages` and then search for presence of `HOTFIX` there.

Reply:
Modified search using `HOTFIX`. 
I found few issues where hot fixes delivered using RPM which not included `HOTFIX` string in release. I am not sure what is the reason behind that but in such cases, this check will fail to find out hot fixes applied using RPM.
Any suggestions on this?